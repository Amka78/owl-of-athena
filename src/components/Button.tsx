//#region Import Modules
import React, { FunctionComponent } from "react";
import { Button as PaperButton } from "react-native-paper";
import { Colors, Dimens, Fonts, Layout } from "../constants";
import { TextStyle, ViewStyle } from "react-native";
//#endregion

//#region Types
export type ButtonProps = {
    children: string;
    disabled?: boolean;
    onPress?: () => void;
};
//#endregion

//#region Component
export const Button: FunctionComponent<ButtonProps> = (props: ButtonProps) => {
    return (
        <PaperButton
            disabled={props.disabled !== undefined ? props.disabled! : false}
            onPress={props.onPress}
            mode={"contained"}
            labelStyle={labelStyle}
            style={containerStyle}
        >
            {props.children}
        </PaperButton>
    );
};
//#endregion

//#region Styles
const labelStyle: TextStyle = {
    color: Colors.navy,
    fontFamily: Fonts.primarySemiBold,
    fontSize: Dimens.button_text_size,
};
const containerStyle: ViewStyle = {
    backgroundColor: Colors.teal,
    borderRadius: Dimens.button_radius,
    marginBottom: Dimens.button_margin_bottom,
    height: Dimens.button_height,
    width: Layout.window.fixedWidth - Dimens.content_margin_horizontal * 2,
};
//#endregion
