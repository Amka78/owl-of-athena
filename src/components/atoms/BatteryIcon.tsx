//#region Import Modules
import React, { FunctionComponent } from "react";

import { Colors, Dimens } from "../../constants";
import { TemplateIcon, TemplateIconProps } from "./TemplateIcon";
//#endregion

//#region Types
export type BatteryIconProps = Omit<TemplateIconProps, "name" | "color"> & {
    batteryLevel: number;
    isUSBConnected: boolean;
};
//#endregion

//#region Component
export const BatteryIcon: FunctionComponent<BatteryIconProps> = (
    props: BatteryIconProps
) => {
    let batteryLevel = props.batteryLevel;
    if (props.batteryLevel === undefined || props.batteryLevel === null) {
        batteryLevel = 0;
    }
    let batteyIconSuffix = "";
    if (props.isUSBConnected) {
        batteyIconSuffix = "-charging";
    }

    let batterIcon = `battery${batteyIconSuffix}`;
    let color = Colors.aurora_connected;

    if (batteryLevel < 100) {
        if (batteryLevel <= 10) {
            color = Colors.red;
            batterIcon = "battery-alert";
        } else {
            batterIcon = `battery${batteyIconSuffix}-${batteryLevel
                .toString()
                .substring(0, 1)}0`;
        }
    }
    return (
        <TemplateIcon
            {...props}
            name={batterIcon}
            color={color}
            size={props.size ? props.size : Dimens.menu_icon_size}
        ></TemplateIcon>
    );
};
//#endregion
