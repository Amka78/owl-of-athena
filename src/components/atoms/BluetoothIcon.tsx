//#region Import Modules
import React, { FunctionComponent } from "react";
import { IconProps } from "react-native-vector-icons/Icon";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";
import { ConnectionStates } from "../../sdk";
import { Colors } from "../../constants";
//#endregion

//#region Types
export type BluetoothIconProps = Omit<IconProps, "name" | "color"> & {
    connectionStates: ConnectionStates;
};
//#endregion

//#region Component
export const BluetoothIcon: FunctionComponent<BluetoothIconProps> = (
    props: BluetoothIconProps
) => {
    let stateIcon = "bluetooth-off";
    let stateColor = Colors.white;
    if (props.connectionStates === ConnectionStates.CONNECTED) {
        stateIcon = "bluetooth-connect";
        stateColor = Colors.aurora_connected;
    } else if (props.connectionStates === ConnectionStates.BUSY) {
        stateIcon = "bluetooth-transfer";
        stateColor = Colors.purple;
    } else if (
        props.connectionStates === ConnectionStates.CONNECTING ||
        props.connectionStates === ConnectionStates.DISCONNECTING ||
        props.connectionStates === ConnectionStates.IDLE ||
        props.connectionStates === ConnectionStates.INIT ||
        props.connectionStates === ConnectionStates.RECONNECTING
    ) {
        stateIcon = "bluetooth-settings";
        stateColor = Colors.aurora_connected;
    }
    return (
        <MaterialCommunityIcons
            {...props}
            color={stateColor}
            name={stateIcon}
        ></MaterialCommunityIcons>
    );
};
//#endregion
