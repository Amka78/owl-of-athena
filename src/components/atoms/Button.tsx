//#region Import Modules
import React, { FunctionComponent } from "react";
import { TextStyle, ViewStyle } from "react-native";
import { Button as PaperButton, useTheme } from "react-native-paper";

import { Dimens, Fonts, Layout, ThemeType } from "../../constants";
//#endregion

//#region Types
export type ButtonProps = {
    children: string;
    disabled?: boolean;
    onPress?: () => void;
    labelStyle?: TextStyle;
    style?: ViewStyle;
};
//#endregion

//#region Component
export const Button: FunctionComponent<ButtonProps> = (props: ButtonProps) => {
    const theme: ThemeType = useTheme();

    return (
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        <PaperButton
            {...props}
            disabled={props.disabled !== undefined ? props.disabled! : false}
            mode={"contained"}
            labelStyle={[labelStyle, props.labelStyle]}
            style={[containerStyle, props.style]}
            theme={theme}
        >
            {props.children}
        </PaperButton>
    );
};
//#endregion

//#region Styles
const labelStyle: TextStyle = {
    fontFamily: Fonts.primarySemiBold,
    fontSize: Dimens.button_text_size,
};
const containerStyle: ViewStyle = {
    borderRadius: Dimens.button_radius,
    marginBottom: Dimens.button_margin_bottom,
    height: Dimens.button_height,
    width: Layout.window.fixedWidth - Dimens.content_margin_horizontal * 2,
};
//#endregion
