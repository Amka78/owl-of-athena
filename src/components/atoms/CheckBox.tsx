//#region Import Modules
import React, { FunctionComponent } from "react";
import { Checkbox as PaperCheckBox, useTheme } from "react-native-paper";
//#endregion

//#region Types
export type CheckBoxProps = {
    color?: string;
    disabled?: boolean;
    status: "checked" | "unchecked" | "indeterminate";
    uncheckedColor?: string;
    onPress?: () => void;
};
//#endregion

//#region Component
export const CheckBox: FunctionComponent<CheckBoxProps> = (
    props: CheckBoxProps
) => {
    const theme = useTheme();
    return (
        <PaperCheckBox
            {...props}
            color={props.color ? props.color : undefined}
            disabled={props.disabled}
            status={props.status}
            onPress={props.onPress}
            uncheckedColor={
                props.uncheckedColor ? props.uncheckedColor : undefined
            }
            theme={theme}
        ></PaperCheckBox>
    );
};
//#endregion
