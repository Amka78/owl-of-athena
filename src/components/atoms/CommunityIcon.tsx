//#region Import Modules
import React, { FunctionComponent } from "react";

import { TemplateIcon, TemplateIconProps } from "./TemplateIcon";
//#endregion

//#region Types
export type CommunityIconProps = Omit<TemplateIconProps, "name">;
//#endregion

//#region Component
export const CommunityIcon: FunctionComponent<CommunityIconProps> = (
    props: CommunityIconProps
) => {
    return <TemplateIcon {...props} name={"account-check"}></TemplateIcon>;
};
//#endregion
