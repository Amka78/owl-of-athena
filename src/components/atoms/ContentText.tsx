//#region Import Modules
import React, { FunctionComponent } from "react";
import { Text, TextStyle } from "react-native";
import { Dimens, Fonts } from "../../constants";
import { ThemeType } from "../../constants/Theme";
import { useTheme } from "react-native-paper";
//#endregion

//#region Types
export type ContentTextProps = {
    children: string;
    onPress?: () => void;
    style?: TextStyle;
};
//#endregion

//#region Component
export const ContentText: FunctionComponent<ContentTextProps> = (
    props: ContentTextProps
) => {
    const theme: ThemeType | undefined = useTheme();

    return (
        <Text
            style={[textStyle, { color: theme?.colors?.accent }, props.style]}
            onPress={props.onPress}
        >
            {props.children}
        </Text>
    );
};
//#endregion

//#region Styles
const textStyle: TextStyle = {
    fontFamily: Fonts.primaryRegular,
    fontSize: Dimens.content_text_size,
    flex: 1,
};
//#endregion
