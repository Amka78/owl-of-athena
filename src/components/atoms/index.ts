//#region Import Moddules
import { BatteryIcon } from "./BatteryIcon";
import { BluetoothIcon } from "./BluetoothIcon";
import { Button } from "./Button";
import { CheckBox } from "./CheckBox";
import { CommunityIcon } from "./CommunityIcon";
import { ContentText } from "./ContentText";
import { ContentTitle } from "./ContentTitle";
import { DeleteIcon } from "./DeleteIcon";
import { ErrorText } from "./ErrorText";
import { FilterIcon } from "./FilterIcon";
import { FlatButton } from "./FlatButton";
import { FlexSpacer } from "./FlexSpacer";
import { HomeIcon } from "./HomeIcon";
import { EditIcon } from "./EditIcon";
import { InfoIcon } from "./Infocon";
import { InlineTimePicker } from "./InlineTimePicker";
import { LeftSideButton } from "./LeftSideButton";
import { ListItem } from "./ListItem";
import { LogoutIcon } from "./LogoutIcon";
import { MenuContainer } from "./MenuContainer";
import { MenuIcon } from "./MenuIcon";
import { OfficialIcon } from "./OfficialIcon";
import { PrivateIcon } from "./PrivateIcon";
import { ProfilesIcon } from "./ProfilesIcon";
import { RefreshIcon } from "./RefreshIcon";
import { ScrollableList } from "./ScrollableList";
import { SessionsIcon } from "./SessionsIcon";
import { SettingsIcon } from "./SettingsIcon";
import { StandardView } from "./StandardView";
import { StarIcon } from "./StarIcon";
import { TextBox } from "./TextBox";
import { TimeView } from "./TimeView";
import { UpdateSnackBar } from "./UpdateSnackBar";
//#endregion

//#region Export
export {
    BatteryIcon,
    Button,
    BluetoothIcon,
    CommunityIcon,
    DeleteIcon,
    FilterIcon,
    TimeView,
    CheckBox,
    ContentText,
    ContentTitle,
    ErrorText,
    FlexSpacer,
    HomeIcon,
    EditIcon,
    InfoIcon,
    InlineTimePicker,
    LeftSideButton,
    ListItem,
    LogoutIcon,
    MenuIcon,
    MenuContainer,
    FlatButton,
    OfficialIcon,
    ProfilesIcon,
    ScrollableList,
    SessionsIcon,
    SettingsIcon,
    StarIcon,
    PrivateIcon,
    RefreshIcon,
    TextBox,
    StandardView,
    UpdateSnackBar,
};
//#endregion
