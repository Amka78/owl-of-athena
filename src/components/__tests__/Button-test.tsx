//#region Import Modules
import "jest-enzyme";
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { TestHelper } from "../../utils/TestHelper";
import { Button, ButtonProps } from "../atoms/Button";
//#endregion

//#region Tests
let component: ShallowWrapper<ButtonProps, unknown, unknown>;

describe("Button UnitTest", () => {
    it("renders correctly", () => {
        component = TestHelper.createMock(<Button>{"test"}</Button>);

        expect(TestHelper.toJson(component)).toMatchSnapshot();
    });

    /*
    it("can overide supportMultiligualState to false.", async () => {
        Message.setLocale("ja-JP");

        const props: IButtonProps = {
            title: "保存"
        };

        component = TestHelper.createMock(<Button {...props} />);
        expect(component.props().title).toBe("保存");
    });

    it("can overide supportMultiligualState to true.", async () => {
        Message.setLocale("ja-JP");

        const props: IButtonProps = {
            localizedTitle: "save"
        };

        component = TestHelper.createMock(<Button {...props} />);
        expect(component.props().title).toBe("保存する。");
    });

    it("can override props", () => {
        Message.setLocale("ja-JP");
        const props: IButtonProps = {
            buttonStyle: {
                backgroundColor: "red",
                borderColor: "white",
                borderRadius: 6,
                borderWidth: 0
            },
            containerStyle: { marginTop: 30, width: "100%" },
            title: "save",
            titleStyle: { fontWeight: "800", backgroundColor: "red" }
        };

        component = TestHelper.createMock(<Button {...props} />);

        // @ts-ignore
        expect(component.props().titleStyle!.fontWeight).toBe("800");
        // @ts-ignore
        expect(component.props().titleStyle!.backgroundColor).toBe("red");
        // @ts-ignore
        expect(component.props().containerStyle!.marginTop).toBe(30);
        // @ts-ignore
        expect(component.props().containerStyle!.width).toBe("100%");
        // @ts-ignore
        expect(component.props().buttonStyle!.backgroundColor).toBe("red");
        // @ts-ignore
        expect(component.props().buttonStyle!.borderColor).toBe("white");
        // @ts-ignore
        expect(component.props().buttonStyle!.borderRadius).toBe(6);
        // @ts-ignore
        expect(component.props().buttonStyle!.borderWidth).toBe(0);
    });
    */
});
//#endregion
