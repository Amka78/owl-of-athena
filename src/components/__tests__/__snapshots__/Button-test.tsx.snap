// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UnitTest renders correctly 1`] = `
<withTheme(Button)
  disabled={false}
  labelStyle={
    Object {
      "color": "#070724",
      "fontFamily": "calibre_app_semibold",
      "fontSize": 24,
    }
  }
  mode="contained"
  style={
    Object {
      "backgroundColor": "#00ffe6",
      "borderRadius": 16,
      "height": 50,
      "marginBottom": 16,
      "width": 654,
    }
  }
>
  test
</withTheme(Button)>
`;
